### PR TITLE
rework yatai setup flow

### PR DESCRIPTION
**Pull request by yubozhao**
Yatai BE:
* In self-host mode, will NOT automatically create admin/org/cluster
* Add `/setup` api endpoint, It takes token, username, email, password and create initial admin user, org, and cluster.

Yatai FE:
* Add setup page.

### CODE REVIEW

File: api-server/controllers/controllersv1/self_hosted.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+			Count: utils.UintPtr(1),
+		},
+		Order: utils.StringPtr("id ASC"),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "list organizations")
+	}
+	if total > 0 {
+		return nil, errors.New("default organization already exists")
+	}
```
Comment (yubozhao) [inline api-server/controllers/controllersv1/self_hosted.go:77]: if total == 1, will set organization to that

File: api-server/controllers/controllersv1/self_hosted.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+			Count: utils.UintPtr(1),
+		},
+		Order: utils.StringPtr("id ASC"),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "list users")
+	}
+	if total > 0 {
+		return nil, errors.New("default admin user already exists")
+	}
```
Comment (yubozhao) [inline api-server/controllers/controllersv1/self_hosted.go:64]: if greater, get the user and compare with the schema

File: api-server/controllers/controllersv1/self_hosted.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+		OrganizationId: defaultOrg.ID,
+		Role:           modelschemas.MemberRoleAdmin,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "create default organization member")
+	}
+
+	// create default cluster
+	var defaultCluster *models.Cluster
+	defaultCluster, err = services.ClusterService.Create(ctx, services.CreateClusterOption{
```
Comment (yubozhao) [inline api-server/controllers/controllersv1/self_hosted.go:126]: update services/clustes.go, in reference of services/cluster_members.go. Use db transaction for create yatai component

File: dashboard/src/components/Header.tsx [hunk 4]
```diff
@@ -208,17 +208,24 @@ export default function Header() {
...
+                            redirect = `${location.pathname}${location.search}`
+                        }
                     } else {
                         redirect = '/'
                     }
                     if (location.pathname !== '/login' && location.pathname !== '/login/') {
-                        window.location.href = `${window.location.protocol}//${
-                            window.location.host
-                        }/login?redirect=${encodeURIComponent(redirect)}`
+                        if (location.pathname !== '/setup' || typeof token !== 'string' || token === '') {
```
Comment (yubozhao) [inline dashboard/src/components/Header.tsx:224]: overcomplicated it

**Comment by yetone** [comment]
@yubozhao I moved all the schemas to a new go mod in the [yatai-schemas](https://github.com/bentoml/yatai-schemas) repo last week, I think you should first update the yatai-schemas repo and restore the changes to the schema directory in this branch, then run `go mod download && go get -u in the `yatai` directory github.com/bentoml/yatai-schemas@latest`.

**Pull request merged**